components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/SCIMConfigProcessorTest.java

```diff
@@ -0,0 +1,91 @@
+        scimConfigProcessor = new SCIMConfigProcessor();
+
+        scimConfigProcessor.properties.put("Name", "aaaaa");
+        scimConfigProcessor.properties.put("Age", "24");
+
+        authenticationSchema = new AuthenticationSchema();
+        authenticationSchema.setName("SCIM");
+        authenticationSchema.setDescription("SCIM2");
+        authenticationSchema.setSpecUri("https://localhost:9443/scim2");
+        authenticationSchema.setDocumentationUri("https://localhost:9443/scim2/docs");
+        authenticationSchema.setType("Authentication");
+        authenticationSchema.setPrimary("true");
+        scimConfigProcessor.authenticationSchemas = new ArrayList<>();
+        scimConfigProcessor.authenticationSchemas.add(authenticationSchema);
+    }
+
+    @AfterMethod public void tearDown() throws Exception {
+    }
+
+    @Test public void testGetProperties() throws Exception {
+        Map<String, String> expected = new HashMap<String, String>();
+        expected.put("Name", "aaaaa");
+        expected.put("Age", "24");
+
+        Map<String, String> map = scimConfigProcessor.getProperties();
+
+        Assert.assertEquals(map.get("Name"), expected.get("Name"));
+        Assert.assertEquals(map.get("Age"), expected.get("Age"));
+    }
+
+    @Test(dataProvider = "facebook-authenticator-all") public void testGetProperty(String property,
+            String expectedResult) throws Exception {
+        Assert.assertEquals(scimConfigProcessor.getProperty(property), expectedResult);
+    }
+
+    @Test public void testGetAuthenticationSchemas() throws Exception {
+        List<AuthenticationSchema> authenticationSchemaList = scimConfigProcessor.getAuthenticationSchemas();
+        for (AuthenticationSchema authenticationSchema1 : authenticationSchemaList) {
+            Assert.assertEquals(authenticationSchema1.getName(), authenticationSchema.getName());
+            Assert.assertEquals(authenticationSchema1.getDescription(), authenticationSchema.getDescription());
+            Assert.assertEquals(authenticationSchema1.getSpecUri(), authenticationSchema.getSpecUri());
+            Assert.assertEquals(authenticationSchema1.getDocumentationUri(),
+                    authenticationSchema.getDocumentationUri());
+            Assert.assertEquals(authenticationSchema1.getType(), authenticationSchema.getType());
+            Assert.assertEquals(authenticationSchema1.getPrimary(), authenticationSchema.getPrimary());
+        }
+    }
+
+    @Test(dataProvider = "filePathProvider", expectedExceptions = CharonException.class) public void testBuildConfigFromFile(
+            String filePath) throws Exception {
+        scimConfigProcessor.buildConfigFromFile(filePath);
+    }
+
+    @Test public void testBuildConfigFromFileHappy() throws Exception {
+        String filePath = "/home/shan/testing_project/identity-inbound-provisioning-scim2/components/org.wso2.carbon.identity.scim2.common/src/test/resources/charon-config-test.xml";
```

Comment:
Use the relative path to the resources directory. Try below.
```
Paths.get(System.getProperty("user.dir"), "src", "test", "resources",
"charon-config-test.xml").toString();
```

Reply:
It worked and it's done.